cpmpy/transformations/normalize.py

```diff
@@ -34,3 +37,118 @@ def unravel(lst, append):
     unravel((cpm_expr,), append)
 
     return newlist
+
+
+def simplify_boolean(lst_of_expr, num_context=False):
+    """
+    removes boolean constants from all CPMpy expressions
+    only resulting boolean constant is literal 'false'
+    - list_of_expr: list of CPMpy expressions
+    """
+
+    newlist = []
+    for expr in lst_of_expr:
+
+        if isinstance(expr, bool):
+            # not sure if this should happen here or at construction time
+            expr = BoolVal(expr)
+
+        if isinstance(expr, BoolVal):
+            newlist.append(int(expr.value()) if num_context else expr)
+
+        elif isinstance(expr, Operator):
+            args = simplify_boolean(expr.args, num_context=not expr.is_bool())
+
+            if expr.name == "or":
```

Comment:
since we are rewriting or/and/impl here anyway:
in a follow-up PR, we should take the optimisations of nested 'and' and implications of ands etc and move them here. That will significantly easy the flatten code.

that will be most powerful after push down negation and decompose is preceding this transfo; so we can do those first

Reply:
Ok, will make a follow-up issue to not forget this.

Reply:
See issue #345